contracts/ownable/src/ZOwnablePK.compact

```diff
@@ -0,0 +1,305 @@
+// SPDX-License-Identifier: MIT
+
+pragma language_version >= 0.16.0;
+
+/**
+ * @module ZOwnablePK
+ * @description A shielded, public key-derived Ownable module.
+ *
+ * `ZOwnablePK` provides a privacy-preserving access control mechanism
+ * for contracts with a single administrative user. Unlike traditional
+ * `Ownable` implementations that store or expose the owner's public key
+ * on-chain, this module stores only a commitment to a hashed identifier
+ * derived from the owner's public key and a secret nonce.
+ *
+ * @notice This module explicitly supports commitments derived from public keys;
+ * however, it may be possible to use contract addresses when contract-to-contract
+ * calls become available. This will be revisited when it is know if/how witnesses
+ * are used from a contract address context.
+ *
+ * @dev Features:
+ * - Obfuscated owner identity: The owner's public key is never revealed on-chain.
+ * - Stateless verification: The contract never needs access to the full public key.
+ * - Built-in support for transfer and renounce functionality.
+ * - Instance-specific salts to prevent cross-contract correlation.
+ * - Deterministic hashing with `persistentHash` to support zero-knowledge verification.
+ *
+ * @dev Commitment structure:
+ * ```
+ * id = H(pk, secretNonce)
+ * commitment = H(id, instanceSalt, counter, "ZOwnablePK:shield:")
+ * ```
+ * The commitment changes on each transfer due to the incrementing `counter`,
+ * providing unlinkability across ownership changes.
+ *
+ * @dev Security Considerations:
+ * - The `secretNonce` must be kept private. Loss of the nonce prevents the
```

Comment:
I think it should also be emphasized that the nonce should be a cryptographically strong random value. Otherwise an attacker can brute force the value and reveal the owner.